main.go

```diff
@@ -371,27 +380,55 @@ func lockFilePresent() bool {
 	return true
 }
 
-func makeReadyHandler(limit *limiter.ConcurrencyLimiter) func(http.ResponseWriter, *http.Request) {
-	return func(w http.ResponseWriter, r *http.Request) {
-		switch r.Method {
-		case http.MethodGet:
-			status := http.StatusOK
+type readiness struct {
+	functionHandler http.Handler
+	endpoint        string
+	lockCheck       func() bool
+	limiter         Limiter
+}
 
-			if atomic.LoadInt32(&acceptingConnections) == 0 || !lockFilePresent() {
-				status = http.StatusServiceUnavailable
-			} else if limit != nil {
-				if limit.Met() {
-					status = http.StatusTooManyRequests
-				}
+func (r *readiness) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case http.MethodGet:
+		status := http.StatusOK
+
+		switch {
+		case atomic.LoadInt32(&acceptingConnections) == 0, !r.lockCheck():
+			status = http.StatusServiceUnavailable
+		case r.limiter.Met():
+			status = http.StatusTooManyRequests
+		case r.endpoint != "":
+			upstream := url.URL{
+				Scheme: req.URL.Scheme,
+				Host:   req.URL.Host,
+				Path:   r.endpoint,
 			}
 
-			w.WriteHeader(status)
-		default:
-			w.WriteHeader(http.StatusMethodNotAllowed)
+			readyReq, err := http.NewRequestWithContext(req.Context(), http.MethodGet, upstream.String(), nil)
+			if err != nil {
+				log.Printf("Error creating readiness request: %s", err)
+				status = http.StatusInternalServerError
+				break
+			}
+
+			// we need to set the raw RequestURI for the function invoker to see our URL path,
+			// otherwise it will just route to `/`, typically this shouldn't be used or set
+			readyReq.RequestURI = r.endpoint
+			readyReq.Header = req.Header.Clone()
+			r.functionHandler.ServeHTTP(w, readyReq)
```

Comment:
Why wouldn't we `http.DefaultClient.Do` this?

Reply:
That requires re-parsing all of the config options to build the correct URL etc. The existing function handler already exists and has all of the desired configuration and URL parsing/construction. 

I think it also matches the semantics of the feature: 

   "during readiness, invoke the function in a certain way"
 
This code matches this semantic directly because we pass down the function invoker (i.e. handler) and then invoke it.

Reply:
@alexellis the doc strings have been added and it also ensures the basic unwrapped handler is used in the readiness check.